bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeConverter.java

```diff
@@ -299,17 +337,33 @@ private static boolean isConvertibleToRecordType(Object sourceValue, BRecordType
             Map.Entry valueEntry = (Map.Entry) object;
             String fieldName = valueEntry.getKey().toString();
 
-            if (targetFieldTypes.containsKey(fieldName)) {
-                if (getConvertibleTypes(valueEntry.getValue(), targetFieldTypes.get(fieldName),
-                                        unresolvedValues).size() != 1) {
+            if (isFromJson) {
+                if (targetFieldTypes.containsKey(fieldName)) {
+                    if (getConvertibleTypesFromJson(valueEntry.getValue(), targetFieldTypes.get(fieldName),
```

Comment:
The only difference in the if and else blocks is the method called right? The arguments are all the same? Instead of the `isFromJson` parameter can we not use something like a functional interface, which I believe is what @rdhananjaya also suggested once.